### DISCUSSION TITLE
Configuration

### DISCUSSION THREAD

**Discussion by kpym** [robpike/ivy#122] [Q&A] [answered]
- Is there a default configuration file for Ivy ?
- Is there a history file ? It looks like the terminal remember the last inputs, but not ivy itself, no ?

**Comment by tcolgate** [comment, marked as answer]
There is no config file at the moment.
There is also no history file, but you can try using rlwrap which will give you shell like line editing, and a history file.

**Comment by kpym** [comment]
@tcolgate Thanks for the suggestion for rlwrap.

For the configuration, I created an alias to `'ivy -f ~/.config/ivy/default.ivy'`.

**Comment by arl** [comment]
I extracted all ivy functions from ivy's `help.go`, so as to be able to autocomplete them when using `rlwrap -f`.
Hope that helps.

https://gist.github.com/arl/eac0a9b9c61cdab04d3e8532566fb5b7

PS: this was done months ago so this might not contain recently added functions. 
It would be nice to be able to recreate that file automatically

**Comment by fzipp** [comment]
@kpym @arl Please have a look at my project [ivy-prompt](https://github.com/fzipp/ivy-prompt/). It might be what you're looking for.

**Reply by arl** [reply]
Oh nice @fzipp ! Indeed that's exactly that. Thank you!
